File: 2021-rabbitmq-jpa-kotlin/core/src/main/code/sample/core/transfer/SingleResponseWrapper.java
package sample.core.transfer;

import lombok.Data;

@Data
public class SingleResponseWrapper<RESPONSE> {

    public static <RESPONSE0> SingleResponseWrapper<RESPONSE0> create() {
        return new SingleResponseWrapper<>();
    }

    private boolean success = true;

    public SingleResponseWrapper<RESPONSE> success(boolean success) {
        this.success = success;
        return this;
    }

    private String lang;

    public SingleResponseWrapper<RESPONSE> lang(String lang) {
        this.lang = lang;
        return this;
    }

    private String code;

    public SingleResponseWrapper<RESPONSE> code(String code) {
        this.code = code;
        return this;
    }

    private String message;

    public SingleResponseWrapper<RESPONSE> message(String message) {
        this.message = message;
        return this;
    }

    private RESPONSE data;

    public SingleResponseWrapper<RESPONSE> data(RESPONSE data) {
        this.data = data;
        return this;
    }
}

File: 2022-declarative/core/src/main/kotlin/sample/core/transfer/SingleResponseWrapper.kt
package sample.core.transfer

class SingleResponseWrapper<RESPONSE> {
    private var success = true
    fun success(success: Boolean): SingleResponseWrapper<RESPONSE> {
        this.success = success
        return this
    }

    private var lang: String? = null
    fun lang(lang: String?): SingleResponseWrapper<RESPONSE> {
        this.lang = lang
        return this
    }

    private var code: String? = null
    fun code(code: String?): SingleResponseWrapper<RESPONSE> {
        this.code = code
        return this
    }

    private var message: String? = null
    fun message(message: String?): SingleResponseWrapper<RESPONSE> {
        this.message = message
        return this
    }

    private var data: RESPONSE? = null
    fun data(data: RESPONSE): SingleResponseWrapper<RESPONSE> {
        this.data = data
        return this
    }

    companion object {
        fun <RESPONSE0> create(): SingleResponseWrapper<RESPONSE0> {
            return SingleResponseWrapper()
        }
    }
}
